association_test.R

```diff
@@ -0,0 +1,62 @@
+#! /usr/bin/env Rscript
+library(argparser)
+library(magrittr)
+argp <- arg_parser("Run association test") %>%
+  add_argument("gds_file", help = "GDS file") %>%
+  add_argument("pheno_file",
+               help = "Phenotype file in annotated dataframe format") %>%
+  add_argument("grm_file", help = "GRM file") %>%
+  add_argument("outcome", help = "Outcome variable name") %>%
+  add_argument("family", help = "Distribution family",
+               default = "gaussian") %>%
+  add_argument("--out_file", help="output file name",
+               default = "assoc.rds") %>%
+  add_argument("--covars",
+               help = "Covariate variable names (space-separated)") %>%
+  add_argument("--variant_id", help = "File with vector of variant IDs") %>%
+  add_argument("--sample_id", help = "File with vector of sample IDs")
+
+argv <- parse_args(argp)
+
+library(SeqArray)
+library(SeqVarTools)
+library(Biobase)
+library(GENESIS)
+
+gds <- seqOpen(argv$gds_file)
+pheno <- readRDS(argv$pheno_file)
+
+if (!is.na(argv$variant_id)) {
+  variant_id <- readRDS(argv$variant_id)
+} else {
+  variant_id <- NULL
+}
+if (!is.na(argv$variant_id)) {
+  sample_id <- readRDS(argv$sample_id)
+} else {
+  sample_id <- NULL
+}
+
+if (!is.na(argv$covars)) {
+  covars <- strsplit(argv$covars, " ") %>% unlist
+} else {
+  covars <- NULL
+}
+
+gds.id <- seqGetData(gds, "sample.id")
+seqData <- SeqVarData(gds, sampleData = pheno)
+seqSetFilter(gds, variant.id = variant_id, sample.id = sample_id)
+iterator <- SeqVarBlockIterator(seqData, verbose=TRUE)
+
+grm <- readRDS(argv$grm_file)
+nullmod <- fitNullModel(pheno, outcome = argv$outcome, covars = covars,
```

Comment:
@smgogarten FYI I was running into trouble trying to run this script because when I made my annotated dataframe, I passed it a tibble instead of a vanilla R dataframe. `fitNullModel()` doesn't like this because you can't set rownames on a tibble. I noticed a few other packages on bioconductor that referenced this, but wasn't sure if it's a known issue for GENESIS

Reply:
I hadn't encountered this before, but just filed an issue in GENESIS: https://github.com/UW-GAC/GENESIS/issues/40